src/ensime-client-startup.ts

```diff
@@ -0,0 +1,47 @@
+import fs = require('fs');
+import path = require('path');
+import loglevel = require('loglevel');
+import chokidar = require('chokidar');
+import utils = require('./server-startup-utils');
+import {DotEnsime} from './types';
+
+
+const log = loglevel.getLogger('ensime.startup');
+const createClient = require('./client');
+
+export type ServerStarter = (dotEnsime: DotEnsime, callback: (pid: string) => void) => void
+
+//  Start an ensime client given path to .ensime. If server already running, just use, else startup that too.
+export default function(serverStarter: ServerStarter) {
+  return function(parsedDotEnsime, generalHandler: (msg: string) => any, callback: (client: any) => any) {
+     
+    const httpPortFilePath = parsedDotEnsime.cacheDir + path.sep + "http";
+
+    if(fs.existsSync(httpPortFilePath)) {
+      // server running, no need to start
+      const httpPort = utils.removeTrailingNewline(fs.readFileSync(httpPortFilePath).toString())
+      createClient(httpPort, generalHandler).then(callback)
```

Comment:
Well, I would actually like a way to kill it anyways - say of Atom died on you and left an orphaned Ensime server running. But It's kind of not possible without hackery right now since we don't have the pid unless we started it in this session.